### PR TITLE
bpo-34790: Document how passing coroutines to asyncio.wait() can be confusing.

### PR DESCRIPTION
**Pull request by 1st1**
https://bugs.python.org/issue34790

### CODE REVIEW

**Comment by 1st1** [comment]
cc @elprans

File: Doc/library/asyncio-task.rst [hunk 1]
```diff
@@ -472,14 +472,20 @@ Waiting Primitives
                             return_when=ALL_COMPLETED)
 
    Run :ref:`awaitable objects <asyncio-awaitables>` in the *aws*
-   sequence concurrently and block until the condition specified
+   set concurrently and block until the condition specified
    by *return_when*.
 
    If any awaitable in *aws* is a coroutine, it is automatically
-   scheduled as a Task.
+   scheduled as a Task.  Although passing coroutines objects to
```
Comment (elprans) [inline Doc/library/asyncio-task.rst]: Drop "although".

**Review by elprans — APPROVED** [review]

**Review by asvetlov — APPROVED** [review]
Neat example :)

**Comment by 1st1** [comment]
> Neat example :)

What do you think about deprecating the current wait() design and fixing it in a couple releases?

**Comment by asvetlov** [comment]
I'd like to deprecate passing regular coroutines into `gather()` as well

**Comment by asvetlov** [comment]
Creating a task if a coroutine is passed to a function is too... implicit.

**Comment by 1st1** [comment]
> I'd like to deprecate passing regular coroutines into gather() as well

Radical :)  I agree, it's confusing.  Although maybe we can keep `gather()` as is and just recommend people to use TaskGroups instead.  I believe `gather()` is way more heavily used than `wait()`.

**Comment by asvetlov** [comment]
Yes, people prefer `gather()` to `wait()`.
I'm ok with not changing `gather()` but hiding it behind task groups in the documentation.
On the other hand, depreciation is not harmful. We can do it slowly starting from `PendingDeprecationWarning`

**Comment by 1st1** [comment]
> I'm ok with not changing gather() but hiding it behind task groups in the documentation.

Yep!

> On the other hand, depreciation is not harmful. We can do it slowly starting from PendingDeprecationWarning

It can be very annoying in large codebases though. I think it's a good idea to use `PendingDeprecationWarning` for `wait()` in 3.8.

**Comment by VasiliPupkin256** [comment]
This makes a simple timeouted wait even more complicated than it was already. You should consider adding timeout to wait() functions then.

What we had:

    await asyncio.wait({event.wait()}, timeout)

Now we have two options:

    await asyncio.wait({asyncio.create_task(event.wait())}, timeout)

or

    with contextlib.suppress(asyncio.TimeoutError):
        await asyncio.wait_for(event.wait(), timeout)

**Pull request merged**